### PR TITLE
Add Rocky Linux support to Azure Linux Agent

### PR DESCRIPTION
**Pull request by nazunalika**
https://rockylinux.org

Rocky Linux is an enterprise Linux distribution based on Red Hat Enterprise Linux. This PR adds Rocky Linux to WALinuxAgent.

## Description

Issue # 

---

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] If applicable, the PR references the bug/issue that it fixes in the description.
- [X] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [X] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def get_data_files(name, version, fullname):  # pylint: disable=R0912
     systemd_dir_path = osutil.get_systemd_unit_file_install_path()
     agent_bin_path = osutil.get_agent_bin_path()
 
-    if name in ('redhat', 'centos', 'almalinux', 'cloudlinux'):  # pylint: disable=R1714
+    if name in ('redhat', 'centos', 'almalinux', 'cloudlinux', 'rocky'):  # pylint: disable=R1714
```
Comment (larohra) [inline setup.py:98]: Can you please remove this suppression, I don't think its needed anymore 

```suggestion
    if name in ('redhat', 'centos', 'almalinux', 'cloudlinux', 'rocky'):
```

**Review by larohra — DISMISSED** [review]
LGTM, running automation.
@kevinclark19a can you take a look please

**Comment by larohra** [comment]
@nazunalika could you please rebase your branch with develop again, thanks! 

> This branch is out-of-date with the base branch

File: setup.py (hunk 1 above)
Comment (nazunalika) [reply setup.py]: I removed the pylint portion on this line for you.

**Comment by nazunalika** [comment]
I believe I have rebased on the latest commits from the develop branch. Let me know if I am missing anything. Thank you!

**Review by kevinclark19a — DISMISSED** [review]
LGTM

**Review by narrieta — APPROVED** [review]

**Comment by kevinclark19a** [comment]
Merged this code in #2463.

**Pull request closed without merging**